### PR TITLE
refactor: Add `<Screen />` component from @blockstack/app

### PR DESCRIPTION
**Pull request by kyranjamie**
[Corresponding `blockstack-app` PR](https://github.com/blockstack/blockstack-app/pull/65)

### CODE REVIEW

File: src/react/components/screen/screen-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            onClick={action.onClick}
+            id={action.id}
+            isDisabled={action.disabled}
+            data-test={action.testAttr}
+          >
+            {action.label}
+          </Button>
+        </Box>
+      )
+    ) : null}
```
Comment (aulneau) [inline src/react/components/screen/screen-actions.tsx]: I've been meaning to do this but these could be broken up into smaller component for readability.

File: src/react/components/screen/screen-body.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  title: string;
+  pretitle?: string | React.ElementType;
+  body?: (string | JSX.Element)[];
+}
+
+export const ScreenBody: React.FC<ScreenBodyProps> = ({ title, body, pretitle }) => (
+  <Stack spacing={2} mx={6}>
+    {pretitle && <Pretitle>{pretitle}</Pretitle>}
+    <Title>{title}</Title>
+    <Stack spacing={[3, 4]}>
```
Comment (aulneau) [inline src/react/components/screen/screen-body.tsx]: I would make this Stack conditional based on if body is passed

File: src/react/components/screen/screen-full-width.tsx [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import React from 'react';
+import { Box } from '@blockstack/ui';
+
+export const ScreenFullWidth: React.FC = ({ children }) => <Box mx="-24px">{children}</Box>;
```
Comment (aulneau) [inline src/react/components/screen/screen-full-width.tsx]: As I mentioned somewhere else, I don't think we need this to be a stand alone component.

File: src/react/components/screens/finished.tsx [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
 
-const AppElement = ({
-  name,
-  icon,
-  ...rest
-}: BoxProps & {
+// Add this in your component file
+require('react-dom');
+(window as any).React2 = require('react');
+console.log((window as any).React1 === (window as any).React2);
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply src/react/components/screens/finished.tsx]: To be removed; is the test to see if there are multiple instances of react.

File: src/react/components/screens/intro.tsx [hunk 5]
```diff
@@ -26,33 +28,35 @@ const AppElement = ({
...
-                icon: EncryptionIcon,
-                text: `Keep everything you do in ${name} private with encryption and blockchain`,
-              },
-              {
-                icon: AppsIcon,
-                text: 'Data Vault is easy to set up and free to use with over 300 apps',
-              },
-            ]}
-          />,
+          <ScreenFullWidth>
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply src/react/components/screens/intro.tsx]: I didn't want the margin on Screen, as this is prohibitive to using a header, like on `blockstack-app`. I've added a `fullWidth` prop to this component.

File: src/react/components/screen/screen-actions.tsx (hunk 1 above)
Comment (kyranjamie) [reply src/react/components/screen/screen-actions.tsx]: Totally agree, and will do.  This has quite some changes here & blockstack-app though, so will do in another PR.

Comment (kyranjamie) [reply src/react/components/screen/screen-actions.tsx]: My approach here is to just use `ScreenActions` as a wrapper, with necessary margins, and put the actions directly as children.

Comment (kyranjamie) [reply src/react/components/screen/screen-actions.tsx]: Done.

**Pull request merged**